### PR TITLE
Make global vars/funcs static, where possible

### PR DESCRIPTION
**Pull request by krzydyn**
Make global variables/functions static where possible.
This allows also to link all tests in one binary (like to use in google test framework)

### CODE REVIEW

File: testcases/misc/math/float/main.c [hunk 1]
```diff
@@ -44,47 +44,47 @@ void myexit(int x)
...
-int generate(char *datadir, char *bin_path)
+static int generate(char *datadir, char *bin_path)
 {
 	char *cmdline;
-	char *fmt = "cd %s; %s/%s %s";
+	const char *fmt = "cd %s; %s/%s %s";
 
-	cmdline = malloc(2 * strlen(bin_path) + strlen(datadir) + strlen(GENERATOR) + strlen(fmt));
+	cmdline = (char *)malloc(2 * strlen(bin_path) + strlen(datadir) + strlen(GENERATOR) + strlen(fmt));
 	if (cmdline == NULL)
```
Comment (pevik) [inline testcases/misc/math/float/main.c:88]: If I remember @metan-ucw correctly, casting to `malloc()` is not required.

Also if it's not too hard for you, could you send patches to mailing list? https://github.com/linux-test-project/ltp/wiki#developers-corner

File: testcases/misc/math/float/main.c [hunk 2]
```diff
@@ -120,12 +120,13 @@ int main(int argc, char *argv[])
 		TCID++;
 	else
 		TCID = argv[0];
+	fprintf(stderr, "main %s\n", TCID);
```
Comment (pevik) [inline testcases/misc/math/float/main.c:123]: why this?

File: testcases/misc/math/float/tfloat.h [hunk 3]
```diff
@@ -53,11 +53,11 @@
...
 #define FUNC_GAM               8
 
-extern void * thread_code(void *);
+static void * thread_code(void *);
 
 /* global variables, constants or initialized by main() */
-extern const double EPS; /* 0.1e-300 */
-extern int true, num_threads;
+//extern const double EPS; /* 0.1e-300 */
+//extern int true, num_threads;
```
Comment (pevik) [inline testcases/misc/math/float/tfloat.h:61]: if not needed just delete it

File: testcases/misc/math/float/main.c (hunk 1 above)
Comment (krzydyn) [reply testcases/misc/math/float/main.c:88]: without cast compiler can generate error:
`error: invalid conversion from ‘void*’ to ‘<some type>*’ [-fpermissive]`

File: testcases/misc/math/float/main.c (hunk 2 above)
Comment (krzydyn) [reply testcases/misc/math/float/main.c:123]: removed

File: testcases/misc/math/float/tfloat.h (hunk 3 above)
Comment (krzydyn) [reply testcases/misc/math/float/tfloat.h:61]: deleted

File: testcases/misc/math/float/main.c (hunk 1 above)
Comment (metan-ucw) [reply testcases/misc/math/float/main.c:88]: If that is a case either you are trying to compile C code with C++ compiler or your compiler is broken.

Comment (krzydyn) [reply testcases/misc/math/float/main.c:88]: That is true. I am using C++.

Comment (metan-ucw) [reply testcases/misc/math/float/main.c:88]: Compiling LTP with C++ compiler is not supported as C++ is subtly incompatible with C.

Comment (krzydyn) [reply testcases/misc/math/float/main.c:88]: Ok. I have reverted type casts.

**Comment by metan-ucw** [comment]
You have send a patch to the mailing list, can we please continue there?

**Comment by krzydyn** [comment]
> You have send a patch to the mailing list, can we please continue there?

Yes. But I can find the web link or address to get there.

**Comment by metan-ucw** [comment]
You have send an email to the mailing list and I've replied to it and that's it. The communication on the mailing list happens over email.

**Comment by metan-ucw** [comment]
Moved over to the mailing list.

**Pull request closed without merging**